From 123c3a0379d5aa7002b7dfde0b234e7d85e2822a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 2 Feb 2025 15:09:46 +0000
Subject: [PATCH] fix: Don't abort if no dictionaries found

PyGTKSpellcheck raises NoDictionariesFound error when no dictionaries found on the system. Instead of aborting the whole program, just disable the spell checker in this case.
---
 iotas/editor_text_view.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/iotas/editor_text_view.py b/iotas/editor_text_view.py
index cce3eac..9ecb130 100644
--- a/iotas/editor_text_view.py
+++ b/iotas/editor_text_view.py
@@ -3,7 +3,7 @@ import gi
 gi.require_version("GtkSource", "5")
 from gi.repository import Gdk, Gio, GLib, GObject, Gtk, GtkSource
 
-from gtkspellcheck import SpellChecker
+from gtkspellcheck import SpellChecker, NoDictionariesFound
 
 import locale
 import logging
@@ -244,7 +244,11 @@ class EditorTextView(GtkSource.View):
             language = locale.getdefaultlocale()[0]
             logging.debug(f'Attempting to use locale default spelling language "{language}"')
 
-        self.__spellchecker = SpellChecker(self, language, collapse=False)
+        try:
+            self.__spellchecker = SpellChecker(self, language, collapse=False)
+        except NoDictionariesFound:
+            return
+
         self.spellchecker_enabled = False
         self.__spellchecker.batched_rechecking = True
         if pref_language is not None:
-- 
GitLab

